app/controllers/application_controller.rb

```diff
@@ -1,4 +1,7 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  def after_sign_in_path_for(resource)
+    stored_location_for(resource) || admin_path
```

Comment:
are you sure you need this?

Reply:
i think not, this is leftover of me doing a lot of unnecessary generations and lines of code from some guide, and later i forgot to remove this :/